sentry/src/main/java/io/sentry/util/UrlUtils.java

```diff
@@ -0,0 +1,186 @@
+package io.sentry.util;
+
+import io.sentry.ISpan;
+import io.sentry.protocol.Request;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class UrlUtils {
```

Comment:
I guess we could mark this as `@ApiStatus.Internal` right?

Reply:
Done